### PR TITLE
[Merged by Bors] - chore(Data/Matrix/Invertible): generalize conjugate and transpose lemmas

### PR DESCRIPTION
**Pull request by eric-wieser**
The `conjTranspose` lemmas now work for non-commutative rings.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Matrix/Invertible.lean [hunk 1]
```diff
@@ -53,6 +63,63 @@ protected theorem mul_mul_invOf_self_cancel (A : Matrix m n α) (B : Matrix n n
...
+def invertibleOfInvertibleTranspose [Invertible Aᵀ] : Invertible A where
+  invOf := (⅟(Aᵀ))ᵀ
+  invOf_mul_self := by rw [←transpose_one, ← mul_invOf_self Aᵀ, transpose_mul, transpose_transpose]
+  mul_invOf_self := by rw [←transpose_one, ← invOf_mul_self Aᵀ, transpose_mul, transpose_transpose]
+#align matrix.invertible__of_invertible_transpose Matrix.invertibleOfInvertibleTranspose
+
+/-- Together `Matrix.invertibleTranspose` and `Matrix.invertibleOfInvertibleTranspose` form an
+equivalence, although both sides of the equiv are subsingleton anyway. -/
+@[simps]
+def transposeInvertibleEquivInvertible : Invertible Aᵀ ≃ Invertible A where
```
Comment (ocfnash) [inline Mathlib/Data/Matrix/Invertible.lean:111]: I'm not sure about this equiv. Will it ever be used except for your proof of `isUnit_transpose` below?

**Review by ocfnash — APPROVED** [review]
Thanks!

Personally I'd excise `transposeInvertibleEquivInvertible` but I'll let you make the call.

bors d+

File: Mathlib/Data/Matrix/Invertible.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Data/Matrix/Invertible.lean:111]: Probably not, but this result is one that we already had in the other file, and I've somewhat established this pattern across the matrix inversion bits of the library, even though I admit it's not a very nice one.

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**